src/hotspot/share/runtime/suspendResumeManager.cpp

```diff
@@ -81,22 +81,33 @@ void SuspendResumeManager::set_suspended(bool is_suspend, bool register_vthread_
   AtomicAccess::store(&_suspended, is_suspend);
 }
 
+void SuspendResumeManager::self_suspend(bool register_vthread_SR, JavaThread *current) {
+#if INCLUDE_JVMTI
+  // All required data should be loaded before state is set to _thread_blocked.
+  int64_t id = java_lang_Thread::thread_id(_target->vthread());
+  ThreadBlockInVM tbivm(current);
+  MutexLocker ml(_state_lock, Mutex::_no_safepoint_check_flag);
```

Comment:
Transitioning to the blocked state and acquiring the monitor should still be executed without JVMTI included. We won’t see any issues today because suspend/resume is only used by JVMTI code.